### PR TITLE
C: Basic Output documentation update

### LINKED ISSUE #893: [Edit + Term Entries] C: Basic Output
### Type of Edit (select all that apply)

Fix grammar, spelling, Clarify unclear content and/or terms, Add new content (definitions, codeblocks, etc.)

### Description (optional)

This entry needs the following edits:

### Line-specific

- Line # 3: meta-`Description` needs to be shortened to (ideally) 150 characters or less.
- Line # 15: Name of concept ("basic console output") needs to be **in bold**.
- Line # 19: Replace "typically" with "commonly".

### Other

The following sections in `basic-output.md` should be reallocated to separate term entries:

- "Print Character with `putchar()`"
- "Print Strings with `puts()`"
- "Print Formatted with `printf()`"

Create a new `terms/` folder under `c/concepts/basic-output/` along with the following folder-file pairs:

- `basic-output/terms/putchar/putchar.md`
- `basic-output/terms/puts/puts.md`
- `basic-output/terms/printf/printf.md`

For each term entry, be sure to include a "Syntax" section (with a \```pseudo code snippet) and an "Example" section (with a \```c code snippet).

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by TheZal**
### Description

Closes #893.

### Type of Change
- Updating the documentation.

- Shortened the description Line # 3
- Name of concept in bold Line # 15
- Replaced "typically" with "commonly" Line # 19
- Created new entries: `basic-output/terms/putchar/putchar.md` - `basic-output/terms/puts/puts.md` - `basic-output/terms/printf/printf.md`

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/c/concepts/basic-output/basic-output.md [hunk 1]
```diff
@@ -12,79 +13,18 @@ CatalogContent:
...
-```
-
-The output would be:
-
-```shell
-5 is a number
-2 is after 1
-```
-
 ## Format Specifiers
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/basic-output.md:28]: The Format Specifier & Decimal Precision sections should go with the new printf() term now.

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+## Print Formatted with `printf()`
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/printf/printf.md:1]: This is missing the Metadata section at the head of the file. See the metadata section [here](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md). These also need some edits to be stand-alone entries. They need introductory material and Syntax and Example headers. You can look at the updated term entry template [here](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md).

File: content/c/concepts/basic-output/terms/putchar/putchar.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+## Print Character with `putchar()`
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/putchar/putchar.md:1]: See my comment on the `printf()` term entry.

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## Print String with `puts()`
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/puts/puts.md:1]: See my comment on the printf() term entry.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@TheZal, I did a first review and left a few comments. Most important is editing that's required on the new term entries. They need to follow the term entry template [here](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md).

**Comment by TheZal** [comment]
Gonna fix that!

File: content/c/concepts/basic-output/basic-output.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ---
 Title: 'Basic Output'
-Description: 'In the C language, basic console output is achieved using functions from the "standard input output" header file (stdio.h). C itself has no ability to output any information. In order to so, the stdio.h header file must be included in the program, which typically happens at the top of the file: c  Once this is done, the functions included in it can be used. The putchar() function, or "put character" function, takes a single char as an argument and prints it to the stdout file, normally the console. c'
+Description: 'In the C lang the console output is managed by the library stdio.h which has to be included.
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/basic-output.md:3]:
```suggestion
Description: 'In the C language the console output is managed by the library stdio.h which has to be included.
```

File: content/c/concepts/basic-output/basic-output.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
...
+Thanks to that you can use function like putchar() that target the C console.'
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/basic-output.md:4]:
```suggestion
Thanks to that a functions that target the C console, like putchar(), can be used.'
```

Comment (SSwiniarski) [reply content/c/concepts/basic-output/basic-output.md:4]: We should avoid "you" and "we" in docs.

File: content/c/concepts/basic-output/basic-output.md (hunk 1 above)
Comment (SSwiniarski) [reply content/c/concepts/basic-output/basic-output.md:28]: These still need to be removed from this file.

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+While the `puts()` function can only accept and print a single string, the **`printf()`** function can print variables within a string. So more flexibility!
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/printf/printf.md:15]:
```suggestion
The **`printf()`** function can print a string to the console, including variables within the string.
```

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+## Syntax
+
+```pseudo
+printf(string)
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/printf/printf.md:20]:
```suggestion
printf(string, value1, value2 ... valueN)
```

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+The `printf()` function, or "print formatted" function, takes a string with variable placeholders, also known as format specifiers, followed by the values to format in the string.
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/printf/printf.md:23]:
```suggestion
The `printf()` function, or "print formatted" function, takes a string with variable placeholders, also known as format specifiers, followed by any number of values to format in the string.
```

File: content/c/concepts/basic-output/terms/putchar/putchar.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Char'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+The **`putchar()`** function, or "put character" function allow to print a char in the `stdout` file, normally the console
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/putchar/putchar.md:15]:
```suggestion
The **`putchar()`** function, or "put character" function allow to print a char to `stdout`, normally the console
```

File: content/c/concepts/basic-output/terms/putchar/putchar.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+## Syntax
+
+```pseudo
+putchar(char)
+```
+
+the **`putchar()`** takes a single `char` as an argument and prints it.
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/putchar/putchar.md:23]:
```suggestion
The **`putchar()`** takes a single `char` as an argument and prints it.
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  - 'Computer Science'
+Tags:
+  - 'String'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+The **`puts()`** function, or "put string" function, allow to print a char in the `stdout` file, normally the console followed by a newline character.
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/puts/puts.md:15]:
```suggestion
The **`puts()`** function, or "put string" function, allow to print a string to `stdout`, normally the console, followed by a newline character.
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+## Syntax
+
+```pseudo
+putchar(char)
+```
+
+the **`puts()`** takes a `string` as an argument and prints it adding a newline character at the end of it.
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/puts/puts.md:23]:
```suggestion
The **`puts()`** takes a `string` as an argument and prints it, adding a newline character at the end of it.
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+  return 0;
+}
+```
+
+The output would be:
+
+```shell
+Hello World!
+```
+
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/puts/puts.md:41]: The content after this point (the extra output example) should be removed.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@TheZal, I've included some more detailed suggestions after your last edit.

**Comment by TheZal** [comment]
Now should be ok... I hope and sorry!

File: content/c/concepts/basic-output/terms/putchar/putchar.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Char'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+The **`putchar()`** function, or "put character" function allow to print a char to `stdout`, normally the console
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/putchar/putchar.md:15]:
```suggestion
The **`putchar()`** function, or "put character" function prints a char to `stdout`, normally the console
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+  - 'Computer Science'
+Tags:
+  - 'String'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+The **`puts()`** function, or "put string" function, allow to print a string to `stdout`, normally the console, followed by a newline character.
```
Comment (SSwiniarski) [inline content/c/concepts/basic-output/terms/puts/puts.md:15]:
```suggestion
The **`puts()`** function, or "put string" function prints a string to `stdout`, normally the console, followed by a newline character.
```

**Review by SSwiniarski — APPROVED** [review]
@TheZal, two more suggestions an I think this is ready for review two. And no apologies necessary, this is just how the process works.

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+The **`printf()`** function can print a string to the console, including variables within the string.
```
Comment (Dusch4593) [inline content/c/concepts/basic-output/terms/printf/printf.md:15]:
```suggestion
The **`printf()`**, or "print formatted", function can print a string to the console, including variables within the string.
```

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+
+## Syntax
+
+```pseudo
+printf(string, value1, value2 ... valueN)
+```
+
+The `printf()` function, or "print formatted" function, takes a string with variable placeholders, also known as format specifiers, followed by any number of values to format in the string.
```
Comment (Dusch4593) [inline content/c/concepts/basic-output/terms/printf/printf.md:23]:
```suggestion
The `printf()` function takes a string with variable placeholders, also known as format specifiers, followed by any number of values to format in the string.
```

File: content/c/concepts/basic-output/terms/printf/printf.md [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+| `%d`             | double  |
+| `%i`             | integer |
+| `%p`             | pointer |
+| `%s`             | string  |
+
+## Decimal Precision
+
+When using format specifiers for decimal numbers, the number of decimal places to be printed can be specified.
+
+To print 2 digits after the decimal place, `0.2f` can be used in `printf()`:
```
Comment (Dusch4593) [inline content/c/concepts/basic-output/terms/printf/printf.md:60]:
```suggestion
For example, to print two digits after the decimal place with `printf()`, `0.2f` can be used:
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+  - 'Computer Science'
+Tags:
+  - 'String'
+  - 'Console'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+The **`puts()`** function, or "put string" function prints a string to `stdout`, normally the console, followed by a newline character.
```
Comment (Dusch4593) [inline content/c/concepts/basic-output/terms/puts/puts.md:15]:
```suggestion
The **`puts()`**, or "put string", function prints a [string](https://www.codecademy.com/resources/docs/c/strings) to `stdout`, normally the console, followed by a newline character.
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+
+## Syntax
+
+```pseudo
+putchar(char)
```
Comment (Dusch4593) [inline content/c/concepts/basic-output/terms/puts/puts.md:20]:
```suggestion
puts(str)
```

File: content/c/concepts/basic-output/terms/puts/puts.md [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+The **`puts()`** takes a `string` as an argument and prints it, adding a newline character at the end of it.
```
Comment (Dusch4593) [inline content/c/concepts/basic-output/terms/puts/puts.md:23]:
```suggestion
The `puts()` function takes a `str` argument and prints it, adding a newline character at the end of it.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@TheZal Finished with my review and left a few comments/suggestions. 😄

**Comment by TheZal** [comment]
Thank you @Dusch4593 :)

**Review by Dusch4593 — APPROVED** [review]
@TheZal LGTM! 👍🏻

**Comment by Dusch4593** [comment]
Congrats on making your first contribution to Codecademy Docs, @TheZal!!! 🎊 🙌🏻 🎉 🥳

Here are the entry links:

- C: Basic Output - https://www.codecademy.com/resources/docs/c/basic-output
  - `printf()` - https://www.codecademy.com/resources/docs/c/basic-output/printf
  - `putchar()` - https://www.codecademy.com/resources/docs/c/basic-output/putchar
  - `puts()` - https://www.codecademy.com/resources/docs/c/basic-output/puts

I wanted to touch on a few things with you.

## Creating a PR Branch for Docs Contributions

For your next Docs contributions, I strongly recommend you branch away from your fork's `main` branch and make/push your changes from there.

Screen Shot 2022-10-28 at 7 11 22 AM

Part of the reason we discourage this is because if you are working on multiple PRs, you will run the risk of causing merge conflicts in one of your PRs which will inhibit our review process from moving forward.

## Anonymous Credit on Entries

Right now, you are listed as "Anonymous" on the list of credited contributors for these entries. To add your profile, you'll need to add your GitHub username to your Codecademy profile, log out of your Codecademy profile, and back in through your GitHub account.

Screen Shot 2022-10-28 at 7 21 10 AM

**Pull request merged**